YDA-5789: add configuration option to limit the number of iRODS conne…

lwesterhof opened the pull request:
…ctions per user

https://github.com/UtrechtUniversity/yoda-ruleset/pull/468

What is an acceptable default max user connections?

lwesterhof committed:
YDA-5789: add configuration option to limit the number of iRODS connections per user

claravox commented on roles/yoda_rulesets/defaults/main.yml:
```diff
@@ -205,6 +205,7 @@ irods_arb_min_gb_free: 0
 irods_arb_min_percent_free: 5
 
 irods_anonymous_account_permit_addresses: []
+irods_user_max_connections: 3                          # Limit the number of iRODS connections per user
```
When uploading a file in one tab, trying to log in another container tab, and running the main playbook I ran against the max connection limit. Otherwise I didn't hit the limit. Maybe set to 4?

lwesterhof committed:
YDA-5789: set limit for the number of iRODS connections per user to 4

claravox reviewed (approved):
LGTM!

stsnel committed:
YDA-5789: update config docs

Mention that configured limit does not apply to rodsadmin users and
the anonymous account.

stsnel committed:
YDA-5789: add enable setting max. connections

Make the policy for maximum number of connections optional, and disable
it by default, because checking the number of connections has a performance
impact, in particular with interactive CLI use (iCommands, goCommands).

stsnel reviewed (approved):
LGTM :+1:

lwesterhof reviewed (commented):
LGTM
